Create taskhw5.cpp

IT-kakaya opened the pull request.

IT-kakaya committed:
Create taskhw5.cpp

dmitrii-kruglov reviewed (changes requested):
Не работает:

Enter the string:
as as as
Words with the first letter A:

dmitrii-kruglov commented on taskhw5.cpp:
```diff
@@ -0,0 +1,41 @@
+    string str2;
+    str2.clear();
+    if (str1[0] == 'A'){
+        for (int j = 0; j < str1.size() - 1; j++){
```
Код повторяется, либо объедините его, или вынесите в отдельную функцию для вызова в двух местах

IT-kakaya replied:
Здесь разные for (у них разная продолжительность и начинаются перебор они с разных ячеек), а внутри цикла вынести if/else не получится, так как тогда не будет работать break (он обязательно должен быть внутри цикла, для того чтобы и остановить в нужный момент его)

dmitrii-kruglov commented on taskhw5.cpp:
```diff
@@ -0,0 +1,41 @@
+    }
+    for (int i = 1; i < str1.size() - 1; i++){
+        if (str1[i] == 'A' && str1[i-1] == ' '){
+            for (int j = i; j < str1.size() - 1; j++){
```
Цикл внутри цикла оправдан ли? Вы же бежите по одной и той же строке.

IT-kakaya replied:
Оправдан, так как в первом цикле идет перебор каждого символа в поисках слов с буквы А, а внутренний цикл лишь дублирует как раз найденное слово и останавливается с пробелом, в то время как общий цикл идет дальше искать букву А

IT-kakaya committed:
Update taskhw5.cpp

Здравствуйте! В задании было написано выводить слова, начинающие с "А", а Вы вводите с "а". Я добавила в условия строчную букву.

dmitrii-kruglov reviewed (changes requested):
Задача была удвоить слова, а не просто их ваывести:

Enter the string:
fdf asd ASd dsa
Words with the first letter A:
asd ASd

IT-kakaya committed:
Update taskhw5.cpp

IT-kakaya commented:
Теперь программа выводит строку с продублированными словами с буквы А/а
